orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -3253,7 +3253,8 @@ Return true if all elements are true and false otherwise.
               /*min_arity*/ 1)
       .Attr("operator", "Name of ATen operator.", AttributeProto::STRING)
       .Attr("overload_name", "Overload name of ATen operator.", AttributeProto::STRING, false)
-      .TypeConstraint("T", OpSchema::all_tensor_types(), "Allow inputs and outputs to be any kind of tensor.");
+      .TypeConstraint("T", OpSchema::all_tensor_types_with_bfloat(),
```

Comment:
You probably have to use `FixedTypeConstraintForHash` so that the operator hash doesn't change. Refer to the [doc](https://github.com/microsoft/onnxruntime/blob/e0d1d6906a04d69e90c06277c32ccff098077006/onnxruntime/test/providers/kernel_def_hash_test.cc#L4-L55)

Reply:
not sure if this is required for training ops. @edgchen1 can you confirm? Thanks!

Reply:
thanks for checking. it is not required now. a CI build should fail if a hash that we care about changes.
what "we care about" is defined here:
https://github.com/microsoft/onnxruntime/blob/48efeca66cbc7eada72a0c59a87476a75b36aa14/tools/ci_build/github/linux/ort_minimal/build_full_ort_and_create_ort_files.sh#L17-L36
note `--enable_training_ops` and no `--enable_training`

the ATen op is not included:
https://github.com/microsoft/onnxruntime/blob/48efeca66cbc7eada72a0c59a87476a75b36aa14/orttraining/orttraining/training_ops/cpu/cpu_training_kernels.cc#L198-L224